src/pharmpy/modeling/plots.py

```diff
@@ -383,99 +444,65 @@ def _dv_vs_anypred(model, predictions, predcol_name, predcol_descr):
     dv_unit = dvcol.unit
     idname = di.id_column.name
 
-    predictions = predictions[[predcol_name]]
-    df = predictions.join(obs, how='inner').reset_index()
+    if stratify_on is None:
+        predictions = predictions[[predcol_name]]
+        df = predictions.join(obs, how='inner').reset_index()
+        title = f"Observations vs. {predcol_descr}s"
+    else:
+        _validate_strat(model, stratify_on)
+        idv_name = model.datainfo.idv_column.name
+        id_name = model.datainfo.id_column.name
+        if (
+            f'{stratify_on}' not in predictions.columns
+            and f'{stratify_on}' not in predictions.index.names
+        ):
+            newcol = model.dataset[[id_name, idv_name, f'{stratify_on}']].set_index(
+                [id_name, idv_name]
+            )
+            predictions = predictions.join(newcol, how='inner')
+            predictions = predictions[[predcol_name, f'{stratify_on}']]
+        else:
+            predictions = predictions[[predcol_name]]
+        df = predictions.join(obs, how='inner').reset_index()
+        df = _bin_data(df, model, stratify_on, bins)
+        title = f"{stratify_on}"
 
     chart = _grouped_scatter(
         df,
         x=predcol_name,
         y=dv,
         group=idname,
-        title=f"Observations vs. {predcol_descr}s",
+        title=title,
         xtitle=predcol_descr,
         ytitle="Observation",
         xunit=dv_unit,
         yunit=dv_unit,
         tooltip=(idv,),
     )
-
     line = _identity_line(
-        df[predcol_name].min(), df[predcol_name].max(), df[dv].min(), df[dv].max()
+        (min_value := min([df[predcol_name].min(), df[dv].min()])) - abs(min_value) * 10,
```

Comment:
If the offset is to prolong the line a bit it should be moved to the function `_identity_line`. Makes the function easier to call and also makes it easier to change the prolongation.